pages/dashboard/workspaces/[workspaceid]/collections/[collectionid]/settings/creators.vue

```diff
@@ -66,65 +76,131 @@ if (error.value) {
 }
 
 if (data.value) {
-  moduleData.creators = normalizeCreators(data.value);
+  creators.value = normalizeCreators(data.value);
 }
 
-const addCreator = () => {
-  moduleData.creators.push({
+const openAddCreatorDrawer = () => {
+  selectedCreator.value = {
     affiliation: "",
-    creatorIndex: moduleData.creators.length + 1,
+    creatorIndex: creators.value.length + 1,
     creatorName: "",
     familyName: "",
     givenName: "",
     identifier: "",
     identifierType: null,
     nameType: "Personal",
+  };
+
+  drawerAction.value = "Add";
+  showCreatorDrawer.value = true;
+};
+
+const openEditCreatorDrawer = (index: number) => {
+  const creator = creators.value[index];
+
+  if (creator) {
+    selectedCreator.value = {
+      affiliation: creator.affiliation,
+      creatorIndex: creator.creatorIndex,
+      creatorName: creator.creatorName,
+      familyName: creator.familyName,
+      givenName: creator.givenName,
+      identifier: creator.identifier,
+      identifierType: creator.identifierType,
+      nameType: creator.nameType,
+    };
+
+    drawerAction.value = "Edit";
+    showCreatorDrawer.value = true;
+  } else {
+    push.error("Something went wrong");
+  }
+};
+
+const deleteCreator = (index: number) => {
+  const newCreators = creators.value.filter(
+    (creator) => creator.creatorIndex !== index,
+  );
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Ensure re-indexing after deleting a creator

After deleting a creator, it's important to re-index the creatorIndex for each creator to maintain a consistent and gapless sequence. This is especially relevant if the index is used in the UI or for ordering.

```suggestion
const newCreators = creators.value
  .filter((creator) => creator.creatorIndex !== index)
  .map((creator, newIndex) => ({ ...creator, creatorIndex: newIndex }));
```